cpp/ycm/ClangCompleter/Diagnostic.h

```diff
@@ -53,6 +81,8 @@ struct Diagnostic {
   std::string text_;
 
   std::string long_formatted_text_;
+
+  std::vector< FixItChunk > fixits_;
```

Comment:
It could be I suppose. In ```YouCompleteMe::Diagnostic``` I'm just cacheing the chunks returned by libClang (in ```YouCompleteMe::BuildDiagnostic``` (ClangHelpers.cpp)) and actually building the ```YouCompleteMe::FixIt``` structure on-demand. The ```FixIt``` structure is really just a lightweight ```Diagnostic``` suitable for the response to ```GetFixItsForLocationInFile```.

I'm not married to this approach, it just seemed the least amount of code to write at the time. For now, I've updated the comments to clarify the usage, if you feel strongly, happy to change this, e.g. so that ```Diagnostic``` contains 0 or 1 ```FixIt``` objects which is checked by ```GetFixItsForLocationInFile``` (probably a *shudder* std::auto_ptr given our c++98-ish-ness)